docs/system-register.md

```diff
@@ -6,16 +6,81 @@ System.register can be considered as a new module format designed to support the
 
 This format provides support for:
 
-* Live bindings updates, including through reexports, star exports and namespace imports, and any combination of these
-* `import.meta`, including `import.meta.url`
 * Dynamic `import()`
+* `import.meta`, including `import.meta.url` and `import.meta.resolve`
 * Top-level await
+* Live bindings updates, including through reexports, star exports and namespace imports, and any combination of these
 * Circular references including function hoisting (where functions in non-executed modules can be used in circular reference cases)
 
 By ensuring we cover all of these semantics, the guarantee is that if code works in browsers in ES modules, the associated
 System.register code can work with s.js providing a legacy fallback workflow that doesn't randomly break at semantic edge cases.
 
-This format was originally developed out of collaboration in Traceur, and is supported as an output in Babel, TypeScript and Rollup.
+This format was originally developed out of collaboration in Traceur, and is supported as an output in Babel, TypeScript and RollupJS.
+
+**This module format is optimized for semantic equivalence, security and performance.**
+
+The general structure of the format is the following:
+
+```js
+System.register(['dependency'], function (_export, _context) {
+  var dep;
+  return {
+    setters: [function (_dep) {
+      dep = _dep;
+    }],
+    execute: function () {
+      _export({
+        name: 'value'
+      });
+    }
+  };
+});
+```
+
+A global callback is used to fully support CSP policies in browsers.
+
+Setters are used over getters to push live bindings since despite being more verbose they offer better runtime performance and
+better handle circularity cases without deadlock.
+
+#### Dynamic Import
+
+The `_context` object contains `_context.import` function corresponding to the contextual dynamic import for the module.
+
+Any `import()` in an ES module can thus be replaced by `_context.import()` to get a semantically identical behaviour for System modules.
+
+### import.meta
+
+The `_context` object contains a `_context.meta` allowing any use of `import.meta` in the ES module to be converted into a `_context.meta` in the System module for semantic equivalence.
+
+#### import.meta.url: String
+
+`_context.meta.url` provides the full URL to the current module as a String.
+
+This is useful for referencing assets by URL in a way that is supported in many modules environments:
+
+```js
+const assetUrl = new URL('./asset.ext', import.meta.url);
+```
+
+#### import.meta.resolve: (id, parentUrl?) => Promise<String>
```

Comment:
I should have brought this up in the PR, but why was it chosen that resolve would be async instead of synchronous? I rely on the top level `System.resolve` api being synchronous inside of https://github.com/joeldenning/systemjs-webpack-interop

Reply:
See https://github.com/whatwg/html/issues/3871#issuecomment-632368537 for the background discussion on this.

The gist of it is though, while there is still a chance that async may be preferable or useful, implementing it as async is better as it can be changed to sync as a non-breaking change, but the opposite isn't possible.

I'm still not personally sure which route is best, but the point is just that the discussions are still in flux.

Reply:
FWIW I could be convinced on a sync resolver in SystemJS as a follow-up.